src/Elm/Parser/Tokens.elm

```diff
@@ -162,92 +162,69 @@ characterLiteral =
         |. Core.symbol "'"
 
 
-type alias StringLiteralLoopState =
-    { escaped : Bool
-    , parts : List String
-    }
-
-
 stringLiteral : Core.Parser String
 stringLiteral =
     Core.succeed identity
         |. Core.symbol "\""
-        |= Core.loop { escaped = False, parts = [] } stringLiteralHelper
-
-
-stringLiteralHelper : StringLiteralLoopState -> Core.Parser (Step StringLiteralLoopState String)
-stringLiteralHelper s =
-    if s.escaped then
-        escapedCharValue
-            |> Core.map
-                (\v ->
-                    Loop { escaped = False, parts = String.fromChar v :: s.parts }
-                )
-
-    else
-        Core.oneOf
-            [ Core.symbol "\"" |> Core.map (\() -> Done (s.parts |> List.Extra.listReverseThenStringConcatFast))
-            , Core.symbol "\\" |> Core.map (\() -> Loop { escaped = True, parts = s.parts })
-            , Core.succeed
-                (\start ->
-                    \value ->
-                        \end ->
-                            if start == end then
-                                Core.problem "Expected a string character or a double quote"
-
-                            else
-                                Core.succeed (Loop { escaped = s.escaped, parts = value :: s.parts })
-                )
-                |= Core.getOffset
-                |= Core.getChompedString (Core.chompWhile (\c -> c /= '"' && c /= '\\'))
-                |= Core.getOffset
-                |> Core.andThen identity
-            ]
+        |= Core.loop [] stringLiteralHelper
 
 
-type alias MultilineStringLiteralLoopState =
-    { escaped : Bool
-    , parts : List String
-    , counter : Int
-    }
+stringLiteralHelper : List String -> Core.Parser (Step (List String) String)
+stringLiteralHelper partsSoFarReverse =
+    Core.oneOf
+        [ Core.symbol "\"" |> Core.map (\() -> Done (partsSoFarReverse |> List.Extra.listReverseThenStringConcatFast))
+        , Core.succeed (\v -> Loop (String.fromChar v :: partsSoFarReverse))
+            |. Core.symbol "\\"
+            |= escapedCharValue
+        , Core.succeed
+            (\start ->
+                \value ->
+                    \end ->
+                        if start == end then
```

Comment:
Can this condition even be `False`? :thinking:

Reply:
Good eye. I don't think it can be False as well.